components/pdf_preview.jsx

```diff
@@ -167,16 +168,24 @@ export default class PDFPreview extends React.PureComponent {
             const fileDownloadUrl = this.props.fileInfo.link || getFileDownloadUrl(this.props.fileInfo.id);
 
             pdfCanvases.push(
-                <a
-                    key='previewpdfmorepages'
-                    href={fileDownloadUrl}
+                <div
                     className='pdf-max-pages'
+                    key='previewpdfmorepages'
                 >
-                    <FormattedMessage
-                        id='pdf_preview.max_pages'
-                        defaultMessage='Download to read more pages'
-                    />
-                </a>,
+                    <button
+                        className='btn btn-primary'
+                        onClick={(e) => {
```

Comment:
Can you make onClick call back a function on the instance?
```
onClick={this.downloadFile}
```

Reply:
Done.